tsdb/series_partition.go

```diff
@@ -614,6 +648,9 @@ func (c *SeriesPartitionCompactor) compactIndexTo(index *SeriesIndex, seriesN ui
 		return err
 	}
 
+	// Attach stats to compactor.
+	c.Stats = stats
+
 	return nil
 }
```

Comment:
Stats are built using the current index snapshot to calculate tombstones. Orphaned tombstones are calculated based on a bitset to determine if a tombstone's series has already been compacted away in an segment.